tests/api/v2/test_knowledge.py

```diff
@@ -176,6 +181,50 @@ async def test_add_fact(knowledge_webapp, aiohttp_client):
     assert current == response
 
 
+async def test_add_fact_to_operation(knowledge_webapp, aiohttp_client, test_operation, setup_empty_operation):
+    client = await aiohttp_client(knowledge_webapp)
+
+    fact_data = {
+        'trait': 'demo',
+        'value': 'test',
+        'source': test_operation['id']
+    }
+    resp = await client.post('/facts', json=fact_data, headers=headers)
+    data = await resp.json()
+    response = data['added']
+    assert len(response) == 1
+    assert response[0]['trait'] == 'demo'
+    assert response[0]['value'] == 'test'
+    assert response[0]['source'] == test_operation['id']
+
+    tmp = await client.get('/facts', json=fact_data, headers=headers)
+    cur = await tmp.json()
+    current = cur['found']
+    assert current == response
+    data_svc = BaseService.get_service('data_svc')
+    file_svc = BaseService.get_service('file_svc')
+    matched_operations = await data_svc.locate('operations', {'id': test_operation['id']})
+    report = await matched_operations[0].report(file_svc, data_svc)
+    assert response[0] in report['facts']
+
+
+async def test_add_fact_to_finished_operation(knowledge_webapp, aiohttp_client, setup_finished_operation,
+                                              finished_operation_payload):
+    client = await aiohttp_client(knowledge_webapp)
+    op_id = finished_operation_payload['id']
+    matched_operations = await BaseService.get_service('data_svc').locate('operations', {'id': op_id})
+    assert await matched_operations[0].is_finished()
+
+    fact_data = {
+        'trait': 'demo',
+        'value': 'test',
+        'source': op_id
+    }
+    resp = await client.post('/facts', json=fact_data, headers=headers)
+    data = await resp.json()
+    assert 'error' in data
```

Comment:
can we also assert that we got back `InvalidOperationStateError("Cannot add fact to finished operation.")`?